service/dap/server.go

```diff
@@ -95,16 +95,28 @@ var defaultArgs = launchAttachArgs{
 	showGlobalVariables: false,
 }
 
-// DefaultLoadConfig controls how variables are loaded from the target's memory, borrowing the
-// default value from the original vscode-go debug adapter and rpc server.
+// DefaultLoadConfig controls how variables are loaded from the target's memory,
+// borrowing the default value from the original vscode-go debug adapter and rpc server.
+// For evaluate requests made in the context of 'repl' or 'variables', they are triggered
+// by explicit human action so we use InteractiveEvaluateLoadConfig to apply larger limits.
 // TODO(polina): Support setting config via launch/attach args or only rely on on-demand loading?
-var DefaultLoadConfig = proc.LoadConfig{
-	FollowPointers:     true,
-	MaxVariableRecurse: 1,
-	MaxStringLen:       64,
-	MaxArrayValues:     64,
-	MaxStructFields:    -1,
-}
+var (
+	DefaultLoadConfig = proc.LoadConfig{
+		FollowPointers:     true,
+		MaxVariableRecurse: 1,
+		MaxStringLen:       64,
+		MaxArrayValues:     64,
+		MaxStructFields:    -1,
+	}
+
+	InteractiveEvaluateLoadConfig = proc.LoadConfig{
```

Comment:
These are **huge**. Consider what happens with this:
```
v := make([]interface{}, 4096)
for i := range v {
    v[i] = &v
}
```

Reply:
Point taken. Now applies only when the final value type is string.